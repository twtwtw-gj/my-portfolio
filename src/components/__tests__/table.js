import React from 'react';
import renderer from 'react-test-renderer';

import Table from '../table';

describe('Table', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Table columns={[]} data={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with data', () => {
    const tree = renderer
      .create(
        <Table
          columns={[
            { Header: 'Col1', accessor: 'col1' },
            { Header: 'Col2', accessor: 'col2' },
          ]}
          data={[
            {
              col1: '1-col1',
              col2: '1-col2',
            },
            {
              col1: '2-col1',
              col2: '2-col2',
            },
          ]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
