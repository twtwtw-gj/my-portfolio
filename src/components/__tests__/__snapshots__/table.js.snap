// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly 1`] = `
<table
  role="table"
>
  <thead />
  <tbody
    role="rowgroup"
  />
</table>
`;

exports[`Table renders correctly with data 1`] = `
<table
  role="table"
>
  <thead>
    <tr
      role="row"
    >
      <th
        colSpan={1}
        role="columnheader"
      >
        Col1
      </th>
      <th
        colSpan={1}
        role="columnheader"
      >
        Col2
      </th>
    </tr>
  </thead>
  <tbody
    role="rowgroup"
  >
    <tr
      role="row"
    >
      <td
        role="cell"
      >
        1-col1
      </td>
      <td
        role="cell"
      >
        1-col2
      </td>
    </tr>
    <tr
      role="row"
    >
      <td
        role="cell"
      >
        2-col1
      </td>
      <td
        role="cell"
      >
        2-col2
      </td>
    </tr>
  </tbody>
</table>
`;
